### PR TITLE
Implemented: support to change box type on in-progress tab(#221)

### LINKED ISSUE #221: Allow Users to Change Box Type for In progress orders
## What is the motivation for adding/enhancing this feature?
App allows users to add boxes for packing products. However, once a box is selected, users are unable to change box type, resulting in limitations when it comes to rate shopping with shipping aggregators.

## What are the acceptance criteria?
Box type must change for added boxes. 

## Can you complete this feature request by yourself?

- [ ] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by sanskar345**
### Related Issues

Closes #221

### Short Description and Why It's Useful

Added box type change functionality in the progress tab in package chips (added ion select to change the box type).

### Screenshots of Visual Changes before/after (If There Are Any)

eee

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/ionic-bopis#contribution-guideline)

### CODE REVIEW

File: src/views/InProgress.vue [hunk 1]
```diff
@@ -801,9 +805,18 @@ export default defineComponent({
       }
       this.addingBoxForOrderIds.splice(this.addingBoxForOrderIds.indexOf(order.orderId), 1)
     },
-    getShipmentPackageNameAndType(shipmentPackage: any, order: any) {
-      // TODO
-      return order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId] ? `Box ${shipmentPackage.packageName} | ${this.boxTypeDesc(order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId][0])}` : ''
+    getShipmentPackageType(shipmentPackage: any, order: any) {
+      const packageType = order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId] ? order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId].find((boxType: string) => boxType === shipmentPackage.shipmentBoxTypeId) : '';
+      if(packageType === undefined){
```
[the remark this answers is not in the document]
Comment (ymaheshwari1) [reply src/views/InProgress.vue:810]: Also if we don't find any value we should display the default value

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/InProgress.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanskar345) [reply src/views/InProgress.vue]: done

File: src/views/InProgress.vue [hunk 2]
```diff
@@ -829,6 +842,11 @@ export default defineComponent({
       await OrderService.printPicklist(picklist.id)
       picklist.isGeneratingPicklist = false;
     },
+    onBoxTypeChange(value: string, shipmentPackage: any, order: any){
```
Comment (ymaheshwari1) [inline src/views/InProgress.vue:845]:
```suggestion
    updateShipmentBoxType(value: string, shipmentPackage: any, order: any) {
```

File: src/views/InProgress.vue [hunk 3]
```diff
@@ -801,9 +805,18 @@ export default defineComponent({
       }
       this.addingBoxForOrderIds.splice(this.addingBoxForOrderIds.indexOf(order.orderId), 1)
     },
-    getShipmentPackageNameAndType(shipmentPackage: any, order: any) {
-      // TODO
-      return order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId] ? `Box ${shipmentPackage.packageName} | ${this.boxTypeDesc(order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId][0])}` : ''
+    getShipmentPackageType(shipmentPackage: any, order: any) {
```
Comment (ymaheshwari1) [inline src/views/InProgress.vue:808]: We can move this logic in the action itself.

File: src/views/InProgress.vue [hunk 4]
```diff
@@ -801,9 +805,18 @@ export default defineComponent({
...
+      let packageType = '';
```
Comment (ymaheshwari1) [inline src/views/InProgress.vue:809]: We can set the default value for the packageType initially and can remove the extra nested if condition.
```suggestion
      let packageType = order.shipmentBoxTypeByCarrierParty[shipmentPackage.carrierPartyId][0];
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Pull request merged**